### PR TITLE
Add HackerNews node

### PR DESCRIPTION
**Pull request by ivov**
Made a very simple test node for the [Hacker News API](https://hn.algolia.com/api).

Ran tests and linter successfully.

### CODE REVIEW

File: packages/nodes-base/nodes/HackerNews/HackerNews.node.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+		description: 'Consume Hacker News API',
+		defaults: {
+			name: 'Hacker News',
+			color: '#772244',
+		},
+		inputs: ['main'],
+		outputs: ['main'],
+		properties: [
+			// ----------------------------------
+			//         Operation
```
Comment (erin2722) [inline packages/nodes-base/nodes/HackerNews/HackerNews.node.ts:26]: Nice commenting!! 😄

File: packages/nodes-base/nodes/HackerNews/HackerNews.node.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	};
+
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		const returnData: IDataObject[] = [];
+
+		const operation = this.getNodeParameter('operation', 0) as string;
+
+		for (let i = 0; i < items.length; i++) {
```
Comment (erin2722) [inline packages/nodes-base/nodes/HackerNews/HackerNews.node.ts:140]: Why do you use a loop here? What is the case in which `i` is not 0? I might totally be mistaken on how this is working, but just a question.

File: packages/nodes-base/nodes/HackerNews/HackerNews.node.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+				endpoint = `users/${this.getNodeParameter('username', i)}`;
+			} else if (operation === 'query') {
+				const tags = this.getNodeParameter('tags', i) as string[];
+				qs = {
+					query: this.getNodeParameter('keyword', i),
+					tags: tags ? tags.join() : ''
+				};
+				endpoint = 'search?';
+			} else {
+				throw new Error(`The operation "${operation}" is not known!`);
```
Comment (erin2722) [inline packages/nodes-base/nodes/HackerNews/HackerNews.node.ts:157]: Maybe use a try-catch block instead here

Comment (ivov) [reply packages/nodes-base/nodes/HackerNews/HackerNews.node.ts:157]: Thanks! Can we think of a benefit of catching when it's only to rethrow? I agree with [this opinion](https://stackoverflow.com/a/42171508) in it makes sense only when you have additional work to do in the catch block.

File: packages/nodes-base/nodes/HackerNews/HackerNews.node.ts (hunk 1 above)
Comment (ivov) [reply packages/nodes-base/nodes/HackerNews/HackerNews.node.ts:26]: Thanks! I did it mostly to avoid losing my place.

**Comment by ivov** [comment]
Definitely, I'll write up how to find article IDs when I do the docs for this node next.

File: packages/nodes-base/nodes/HackerNews/HackerNews.node.ts (hunk 3 above)
Comment (erin2722) [reply packages/nodes-base/nodes/HackerNews/HackerNews.node.ts:157]: That's a good point! I do think that the way you wrote this code makes it very readable-- was more thinking about conventions in the codebase? But I think leaving it is definitely good!

Comment (ivov) [reply packages/nodes-base/nodes/HackerNews/HackerNews.node.ts:157]: Agreed, it's something we should improve on in the codebase.

**Comment by ivov** [comment]
Submitted to maintainers for approval: https://github.com/n8n-io/n8n/pull/670

File: packages/nodes-base/nodes/HackerNews/HackerNews.node.ts (hunk 2 above)
Comment (RicardoE105) [reply packages/nodes-base/nodes/HackerNews/HackerNews.node.ts:140]: @erin2722 If the node input has 20 items then `i` will take from 0 to 19.

**Pull request closed without merging**